tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala

```diff
@@ -66,17 +73,100 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
+
+    val wisecondorZJobs = wisecondorRefJobs map {
+      case (sample, jobsAndRefFile) =>
+        val refJobs = jobsAndRefFile._1
+        val refFile = jobsAndRefFile._2
+        val tbiFile = refJobs.collect { case tbi: Tabix => tbi.outputIndex }.head
+        sample -> createWisecondorZScore(sample, refFile, tbiFile)
+    }
+
+    val xhmmZJobs = xhmmRefJobs map {
+      case (sample, jobsAndRefFile) =>
+        sample -> createXhmmZscore(sample, jobsAndRefFile._2)
+    }
+
+    val wisecondorSyncJobs = wisecondorZJobs map {
+      case (sample, job) =>
+        val intersect = new BedtoolsIntersect(this)
+        intersect.input = job.output
+        intersect.intersectFile = xhmmZJobs(sample)._2
+        intersect.output = new File(sample.wisecondorDir, s"${sample.sampleId}.wisecondor.sync.z.bed")
+        sample -> intersect
+    }
+
+    val xhmmSyncJobs = xhmmZJobs map {
+      case (sample, jobsAndFile) =>
+        val intersect = new BedtoolsIntersect(this)
+        intersect.input = jobsAndFile._2
+        intersect.intersectFile = wisecondorZJobs(sample).output
+        intersect.output = new File(sample.xhmmDir, s"${sample.sampleId}.xhmm.sync.z.bed")
+        sample -> intersect
+    }
+
+    val zScoreMergeJobs = samples map {
+      case (_, sample) =>
+        val horizontal = new StouffbedHorizontal(this)
+        val inputs = List(
+          wisecondorSyncJobs(sample).output,
+          xhmmSyncJobs(sample).output
+        )
+        horizontal.inputFiles = inputs
+        horizontal.output = new File(sample.sampleDir, s"${sample.sampleId}.horizontal.bed")
+        sample -> horizontal
+    }
+
+    val windowStouffJobs = zScoreMergeJobs map {
+      case (sample, horizontal) =>
+        val subMap = stouffWindowSizes map { size =>
+          val windowDir = new File(sample.sampleDir, s"window_$size")
+          val vertical = new StouffbedVertical(this)
+          vertical.inputFiles = List(horizontal.output)
+          vertical.output = new File(windowDir, s"${sample.sampleId}.window_$size.z.bed")
+          vertical.windowSize = size
+          size -> vertical
+        }
+        sample -> subMap.toMap
     }
 
-    (wisecondorRefJobs.toList ::: xhmmRefJobs.toList).flatten.foreach(job => add(job))
+    val thresholdJobs = windowStouffJobs map {
+      case (sample, subMap) =>
+        val threshSubMap = subMap map {
+          case (size, job) =>
+            val windowDir = new File(sample.sampleDir, s"window_$size")
+            val thresholder = new BedThreshold(this)
+            thresholder.input = job.output
+            thresholder.output = Some(new File(windowDir, s"${sample.sampleId}.threshold.bed"))
+            thresholder.threshold = threshold
+            size -> thresholder
+        }
+        sample -> threshSubMap
+    }
+
+    _finalFiles = thresholdJobs map {
+      case (sample, subMap) =>
+        sample -> subMap.values.flatMap(_.output).toList
+    }
+
+    addAll(xhmmRefJobs.values.flatMap(_._1))
```

Comment:
Since it's inside the same method maybe better to move this to the blocks where they belong to?